Vector_Class/Vector.h

```diff
@@ -0,0 +1,191 @@
+#ifndef VECTOR_CLASS_VECTOR_H
+#define VECTOR_CLASS_VECTOR_H
+
+#include <iostream>
+
+template <typename T>
+class Vector
+{
+private:
+    size_t size_;
+    size_t capacity_;
+    T* arr = nullptr;
+
+    void relocate()
+    {
+        capacity_ *= 2;
+        T* arr2 = new T[capacity_];
+
+        for(int i = 0; i < size_; ++i)
+            arr2[i] = arr[i];
+
+        delete[] arr;
+        arr = arr2;
+    }
+
+
+public:
+
+    explicit Vector(size_t size = 0, T elem = 0) : size_(size), capacity_(size_)
+    {
+        if (size_ == 0 || size_ == 1)
+            capacity_ = 2;
+
+        arr = new T[capacity_];
+
+        for (int i = 0; i < size_; ++i) {
+            arr[i] = elem;
+        }
+    }
+
+    ~Vector()
+    {
+        size_ = 0;
+        capacity_ = 0;
+        delete[] arr;
+    }
+
+
+    Vector(const Vector<T>& other)
+    {
+        delete[] arr;
+        arr = new T[capacity_];
+        size_ = other.size_;
+        capacity_ = other.capacity_;
+
+        for (int i = 0; i < size_; ++i)
+            arr[i] = other.arr[i];
+
+    }
+
+    Vector(Vector<T>&& other)
+    {
+        delete[] arr;
+        size_ = other.size_;
+        capacity_ = other.capacity_;
+        arr = other.arr;
+        other.arr = nullptr;
+    }
+
+    Vector<T> &operator=(const Vector<T> &other)
+    {
+        if (this == &other)
+            return *this;
+
+        delete[] arr;
+        arr = new T[capacity_];
```

Comment:
new array should be created with other.capacity, that is
```
arr = new T[other.capacity];
```

Reply:
👍